File: nhlapi/content.go
package nhlapi

type ContentResponse struct {
	Media      ContentMedia      `json:"media"`
	Highlights ContentHighlights `json:"highlights"`
}

type ContentMedia struct {
	EPG []*ContentEPG `json:"epg"`
}

type ContentEPG struct {
	Title string         `json:"title"`
	Items []*ContentItem `json:"items"`
}

type ContentItem struct {
	Type      string             `json:"type"`
	Playbacks []*ContentPlayback `json:"playbacks"`
}

type ContentPlayback struct {
	Name string `json:"name"`
	URL  string `json:"url"`
}

type ContentHighlights struct {
	Scoreboard ContentScoreboard `json:"scoreboard"`
}

type ContentScoreboard struct {
	Items []*ContentHighlightItems `json:"items"`
}

type ContentHighlightItems struct {
	ID          string             `json:"id"`
	Type        string             `json:"type"`
	Title       string             `json:"title"`
	Blurb       string             `json:"blurb"`
	Description string             `json:"description"`
	Keywords    []*ContentKeyword  `json:"keywords"`
	Playbacks   []*ContentPlayback `json:"playbacks"`
}

type ContentKeyword struct {
	Type  string `json:"type"`
	Value string `json:"value"`
}

File: rust/nhl-highlights/src/nhlapi/content.rs
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct ContentResponse {
    pub media: ContentMedia,
}

#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct ContentMedia {
    pub epg: Vec<ContentEPG>,
}

#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct ContentEPG {
    title: String,
    items: Vec<ContentItem>,
}

#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct ContentItem {
    #[serde(rename = "type")]
    type_: String,
    playbacks: Vec<ContentPlayback>,
}

#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct ContentPlayback {
    name: String,
    url: String,
}
